hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenFreezeAccountHandler.java

```diff
@@ -112,8 +113,8 @@ private TokenRelation validateSemantics(
             @NonNull final WritableTokenRelationStore tokenRelStore)
             throws HandleException {
         // Check that the token exists
-        final var tokenId = op.tokenOrElse(TokenID.DEFAULT);
-        final var tokenMeta = tokenStore.getTokenMeta(tokenId);
+        final var token = TokenHandlerHelper.getIfUsable(op.tokenOrElse(TokenID.DEFAULT), tokenStore);
```

Comment:
Okay, but `op.tokenOrThrow()` will return the token even if it's `deleted`. That's the body of `TokenHandlerHelper.getIfUsable`, it contains a check for `TOKEN_WAS_DELETED` which is needed here.
```java
    public static Token getIfUsable(@NonNull final TokenID tokenId, @NonNull final ReadableTokenStore tokenStore) {
        requireNonNull(tokenId);
        requireNonNull(tokenStore);

        final var token = tokenStore.get(tokenId);
        validateTrue(token != null, INVALID_TOKEN_ID);
        validateFalse(token.deleted(), TOKEN_WAS_DELETED); // <--- we need this check
        validateFalse(token.paused(), TOKEN_IS_PAUSED);
        return token;
    }
```

Another approach could be to remove `TokenHandlerHelper.getIfUsable` from the method and add a simple check like that, but in that case we permit user to `freezeAccount` for `paused` token, which is not right IMO.
```java
    validateFalse(token.deleted(), TOKEN_WAS_DELETED);
```
Adding a second check for `TOKEN_IS_PAUSED` would lead to almost the same body as the `TokenHandlerHelper.getIfUsable`. I think we don't want to copy-paste the same things over the project.

Or am I missing something? If we skip this check, we will be able to `freezeAccount` for already deleted token which is bad behaviour, I guess.

Reply:
Sorry for not being clear. I meant we should use `op.tokenOrThrow()` instead of `op.tokenOrElse(TokenID.DEFAULT)`. I agree that calling `getIfUsable()` is mandatory.

Reply:
Oh, got it.